pkg/provisioner/reconciler/claim-reconciler/reconiler.go

```diff
@@ -202,6 +205,10 @@ func (r *ObjectBucketClaimReconciler) handleProvisionClaim(key client.ObjectKey,
 	ob.Spec.StorageClassName = obc.Spec.StorageClassName
 	ob.Spec.ClaimRef, err = claimRefForKey(key, r.internalClient)
 	ob.SetFinalizers([]string{finalizer})
+	// allow provisioner to modify reclaimPolicy
+	if ob.Spec.ReclaimPolicy == nil {
+		ob.Spec.ReclaimPolicy = options.ReclaimPolicy
```

Comment:
I'm thinking this probably shouldn't be conditional.  Rather than letting provisioners set the policy, we should enforce the one defined by the storage class.  Otherwise the SC isn't representational of the backing OBs.